### ISSUE TITLE
Allow `compose` transformer return arbitrary type

### ISSUE THREAD

**Issue by whitecolor** [staltz/xstream#207]
Currently `compose` requires that returned value was a stream of transformed values. 

What about allowing to return arbitrary type of value?

```ts
const partition = <T>(predicate: (item: T) => boolean) =>
  (stream: Stream<T>) => [
    stream.filter(predicate),
    stream.filter(item => !predicate(item))
  ]

// this would allow such syntax
const [good$, bad$] = source$.compose(partition(filterGood))

// vs 

const [good$, bad$] = partition(filterGood)(source$)
```

Or you see `compose` as strictly Stream to Stream transformation function? (though JS without TS version still allows to do this)

**Comment by staltz** [comment]
@whitecolor This is interesting. I thought it would hurt typings elsewhere, but I gave it a try, and we can easily do this with zero difference for JS users, and no breaking changes for anyone (JS or TS users). It's just a backwards compatible feature. It's an easy win, so why not.

**Comment by staltz** [comment]
It's so easy, I just did it. Released in v10.9

**Comment by whitecolor** [comment]
Great. This allows more flexible operators. BTW what do you is `partition` operator needed in extras? I quite often find it in a need.
